src/number/n_unknown.jl

```diff
@@ -242,8 +242,116 @@ end
 #
 ###############################################################################
 
+mutable struct MutableRingWrapper{S, T} <: Nemo.Ring
+   data::S
+end
+
+mutable struct MutableRingElemWrapper{S, T} <: Nemo.RingElem
+   data::T
+   parent::MutableRingWrapper{S, T}
+end
+
+function promote_rule(::Type{n_unknown{MutableRingElemWrapper{S, T}}}, ::Type{T}) where {T <: Nemo.RingElem, S}
+   return T
```

Comment:
If the return type is `T`, then the promotion system will do
`parent(y)(x) + y`, where
`x isa n_unknown{MutableRingElemWrapper{S, T}}` and `y isa T`. Do we want this? 

I probably told you the wrong thing on Friday, sorry.

Reply:
eh, it could go either way. I am thinking now the unknown mutable wrapper might be better because that is the type of everything for singular.